### PR TITLE
Fixes

### LINKED ISSUE #1: Logic Improvement
There are many pssible ways you can think of.
You can try implementing char by char matching.
You can implement better methods for generating paragraphs.
Handle exceptional cases. The game does not work correctly if the paragraph has contiguous spaces.

### LINKED ISSUE #7: Result analysis after race in impressive way.
You can use graphs, charts or any other way you can think of.

### PR DESCRIPTION
**Pull request by iosdev474**
Fix #1,#2,#3,#4,#5,#7
Added leaderboard instead of current position
fix #7 not in graph but used a table.

### CODE REVIEW

File: Typeracer_Game/src/typeracer_game/GameStarter.java [hunk 1]
```diff
@@ -43,34 +52,95 @@ public void actionPerformed(java.awt.event.ActionEvent evt) {
...
+    Game ob = new Game(nameTextField.getText(),ipTextField.getText(), portTextField.getText());
     this.setVisible(false);
     ob.setVisible(true);
 
 }//GEN-LAST:event_startButtonActionPerformed
 
+    private void serverButtonActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_serverButtonActionPerformed
+        serverLabel.setText("Server Started");
+        serverLabel.paintImmediately(serverLabel.getVisibleRect());
+        obj = new Server();
```
Comment (Rashi-G) [inline Typeracer_Game/src/typeracer_game/GameStarter.java]: It's showing error here. You have defined constructor with two arguments in Server class.

Comment (iosdev474) [reply Typeracer_Game/src/typeracer_game/GameStarter.java]: :p forget to add updated GameStarter.java

**Comment by iosdev474** [comment]
now it is good?

File: Typeracer_Game/src/typeracer_game/GameStarter.java [hunk 2]
```diff
@@ -43,34 +54,113 @@ public void actionPerformed(java.awt.event.ActionEvent evt) {
...
         );
 
         pack();
     }// </editor-fold>//GEN-END:initComponents
 
 private void startButtonActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_startButtonActionPerformed
-// TODO add your handling code here:
-    Game ob = new Game();
+
+    Game ob = new Game(nameTextField.getText(),ipTextField.getText(), portTextField.getText(), Integer.parseInt(clientsTF.getText()));
```
Comment (Rashi-G) [inline Typeracer_Game/src/typeracer_game/GameStarter.java:144]: Again error in arguments of constructor.

Comment (iosdev474) [reply Typeracer_Game/src/typeracer_game/GameStarter.java:144]: what is the error now the arguments are correct 3XString,1XInt

**Pull request closed without merging**